applications/crossbar/src/modules/cb_rates.erl

```diff
@@ -555,3 +617,32 @@ normalize_field(<<"Surcharge">> = K, Surcharge) ->
     {K, wht_util:units_to_dollars(Surcharge)};
 normalize_field(K, V) ->
     {K, V}.
+
+-spec account_rates_db(ne_binary()) -> ne_binary().
+account_rates_db(AccountId) ->
+    FormatedAccountDb = wh_util:format_account_db(AccountId),
+    <<FormatedAccountDb/binary, "-", ?WH_RATES_DB/binary>>.
+
+-spec account_parents(cb_context:context()) -> ne_binaries().
+account_parents(Context) ->
+    AccountId = cb_context:account_id(Context),
+    kz_account:tree(cb_context:doc(crossbar_doc:load(AccountId, Context))).
+
+-spec init_db(ne_binary()) -> 'ok'.
+init_db(DbName) ->
+    case couch_mgr:db_exists(DbName) of
+        'true' -> 'ok';
+        'false' ->
+            _ = couch_mgr:db_create(DbName),
+            couch_mgr:revise_doc_from_file(DbName, 'crossbar', "views/rates.json"),
+            'ok'
+    end.
+
+-spec set_account_db(cb_context:context()) -> cb_context:context().
+set_account_db(Context) ->
+    Db = case account_parents(Context) of
+             [] -> ?WH_RATES_DB;
+             _Parents -> account_rates_db(cb_context:account_id(Context))
+         end,
+    init_db(Db),
```

Comment:
I'd rather detect, when loading the view, if it was a 'not_found' error and init the DB (or check the DB), rather than checking the DB on every request.